api/v1beta2/sparkapplication_types.go

```diff
@@ -407,6 +407,11 @@ type Dependencies struct {
 // SparkPodSpec defines common things that can be customized for a Spark driver or executor pod.
 // TODO: investigate if we should use v1.PodSpec and limit what can be set instead.
 type SparkPodSpec struct {
+	// Template is a pod template that can be used to define the driver or executor pod configurations that Spark configurations do not support.
+	// Spark version >= 3.0.0 is required.
+	// Ref: https://spark.apache.org/docs/latest/running-on-kubernetes.html#pod-template.
+	// +optional
+	Template *corev1.PodTemplateSpec `json:"template,omitempty"`
```

Comment:
Will review the rest of the PR as well, but adding this as a struct rather than a binary/string field increases the number of properties in the resulting CRD by quite a bit. Is there any downside/considerations on the K8s side for very large CRDs?

Reply:
Good catch. I added a kubebuilder marker `+kubebuilder:validation:Schemaless` to mark the pod template field as schemaless to reduce the size of CRD.

Reply:
I'm quite excited for this PR so I gave it a try. Maybe I did something wrong so feel free to ignore me but I got the following error:
```
Error from server (BadRequest): error when creating "<path>": SparkApplication in version "v1beta2" cannot be handled as a SparkApplication: strict decoding error: unknown field "spec.driver.template.metadata", unknown field "spec.driver.template.spec", unknown field "spec.executor.template.metadata", unknown field "spec.executor.template.spec"
```
when running my `kubectl` command. 

Removing `+kubebuilder:validation:Schemaless` and regenerating the CRD solved it for me.